src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessor.java

```diff
@@ -67,9 +68,8 @@ public void process(BlobClient blobClient) {
             .acquireFor(blobClient)
             .ifPresentOrElse(
                 leaseClient -> {
+                    UUID id = envelopeService.createNewEnvelope(containerName, blobName, blobCreationDate);
                     try {
-                        UUID id = envelopeService.createNewEnvelope(containerName, blobName, blobCreationDate);
-
```

Comment:
Why was this line move outside the `try` part?

Reply:
so that we have the envelope ID. without it we can't create events in the new flow (foreign key instead of file name + container name).

Reply:
I guess a failure to create a new envelope is one of the reasons to create an error event. Maybe we should define the id (as null) before the try, create the envelope inside `try` and be prepared that it's null in the `catch` clause?

Reply:
Aaa, sorry, it's the foreign key.